src/index.js

```diff
@@ -150,8 +152,9 @@ class Dotenv {
     // However, if someone targets Node or Electron `process.env` still exists, and should therefore be kept
     // https://webpack.js.org/configuration/target
     if (this.shouldStub({ target, version })) {
```

Comment:
I think `shouldStub` should be made to return true only when prefix is `process.env` (based on the reasoning in the comment above)

Reply:
Thank you! I've made changes, now it stubs only when prefix is `process.env`

Reply:
Yeah, the water is getting a little muddy here. Originally the system implied based on env (which I think is safe). Perhaps the stub can work as is now, but if we see the prefix as manually set we force the stub you defined always.

Reply:
Sorry, I didn't fully understand the intended behaviour when the prefix is manually set. Should we continue to stub `process.env` or the set prefix?

Reply:
To keep this as a non-major version update so people can update without things breaking I suggest we let the system work as it used to, but should they explicitly set the “prefix” property it would always apply the stub with whatever they have requested.

Reply:
Am I getting it right?

```js
// Current behaviour

new Dotenv()
// process.env.MISSING --> "MISSING_ENV_VAR".MISSING

new Dotenv({
  ignoreStub: true
})
// process.env.MISSING --> process.env.MISSING
```
```js
// With set option

new Dotenv({
  prefix: 'META_ENV_'
})
// META_ENV_MISSING --> "MISSING_ENV_VAR".MISSING
// process.env.MISSING --> process.env.MISSING

new Dotenv({
  prefix: 'META_ENV_',
  ignoreStub: true
})
// META_ENV_MISSING --> META_ENV_MISSING
// process.env.MISSING --> process.env.MISSING
```

Reply:
No. We shouldn’t be changing anything about how it works currently and you have changed how the plug-in works out-of-the box (if I am reading this right).

Reading some of these assumptions I’m also worried the original point of the stub has been lost.

the basic requirements of this pr is quite simple:
1. Should support a “prefix” configuration property and should be able to override the existing `process.env.` prefix (which is the default).
2. Stubbing (which was the last major feature introduced to account for all the unique webpack targets) should work as it does today, but if you override the default prefix you should also force the stub (using of course the defined prefix in the configuration).


With that being said, I do think it’s work getting rid of the stub feature and allowing users to just manually set to an empty string to remove the stub for unique targets, but it’s a little more debatable as that also introduces other complexities.

Reply:
The plug-in stubs `process.env` with `"MISSING_ENV_VAR"` out-of-the-box when the env var is missing, I didn't change that, plug-in works as it did before, unless a `prefix` property is set. I've illustrated this with the code example with comment "Current behaviour".

The second example is my understanding of how the plugin should work when the property is set.

I've made the dot explicit in the most recent commit, but I didn't change the bit, which disables stubbing when the prefix is set to other value, than `process.env.`. Currently DefinePlugin can replace full expression or a part of expression, with a dot following it. For us it means that we can't replace every possible user defined prefix, but only those, ending with a dot. That is why I kept this restriction for now.

I can try to explore other possibilities for stubbing custom prefixes, but as I see now, it most likely will involve changes in DefinePlugin.